flink-end-to-end-tests/test-scripts/test_ha_datastream.sh

```diff
@@ -51,6 +51,7 @@ function run_ha_test() {
     # change the pid dir to start log files always from 0, this is important for checks in the
     # jm killing loop
     set_conf "env.pid.dir" "${TEST_DATA_DIR}"
+    set_conf "env.java.opts" "-ea"
```

Comment:
I think we might want to reach consensus about that on a broader scope than just this PR. IMO, `-ea` should be enabled in all tests. I enabled it here because it was relevant for my PR.